R/mediate.R

```diff
@@ -0,0 +1,394 @@
+#' Perform mediation analysis
+#'
+#' \code{getMediation} provides a wrapper of \code{\link[mediation:mediate]{mediate}}
+#' for \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}.
+#'
+#' @param x a \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}.
+#' 
+#' @param outcome The name of the colData variable used as outcome in the model.
+#' 
+#' @param treatment The name of the colData variable used as treatment in the model.
+#'
+#' @param mediator The name of the colData variable used as mediator in the model.
+#'   (default: \code{mediator = NULL})
+#'
+#' @param assay.type The name of the assay used for feature-wise mediation analysis.
+#'   (default: \code{assay.type = NULL})
+#' 
+#' @param dim.type The name of the reducedDim used for component-wise mediation analysis.
+#'   (default: \code{dim.type = NULL})
+#'
+#' @param family A specification for the outcome model link function.
+#'    (default: \code{family = gaussian("identity")})
+#' 
+#' @param covariates Name or list of names of colData variables used as covariates
+#'   in the model. (default: \code{covariates = NULL})
+#' 
+#' @param p.adj.method A single character value for selecting adjustment method
+#'   of p-values. Passed to `p.adjust` function. (default: \code{p.adj.method = "BH"})
+#' 
+#' @param add.metadata TRUE or FALSE, should the model metadata be returned.
```

Comment:
Also mention how it is returned; added to metadata slot of the SE object?

Reply:
Show in examples how that can be then retrieved from the object (if not yet shown)